Brokerages/Backtesting/BacktestingBrokerage.cs

```diff
@@ -214,6 +215,13 @@ public void Scan()
                         continue;
                     }
 
+                    // all order fills are processed on the next bar
+                    if (order.Time == Algorithm.UtcTime)
+                    {
```

Comment:
very smart to implement it like this, maybe just add `order.Type != OrderType.Market` to allow market orders to pass